### PR TITLE
fix missing on_load parameter in custom_404

### PR DESCRIPTION
**Pull request by Lendemor**
### All Submissions:

- [x] Have you followed the guidelines stated in [CONTRIBUTING.md](https://github.com/pynecone-io/pynecone/blob/main/CONTRIBUTING.md) file?
- [x] Have you checked to ensure there aren't any other open [Pull Requests](https://github.com/pynecone-io/pynecone/pulls ) for the desired changed?

### Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] This change requires a documentation update

### Changes To Core Features:

- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you successfully ran tests with your changes locally?

### Description

Add the missing on_load kwargs to the custom 404 method, allowing a default redirect when a page is unknown.

Also make component optional (default to a rx.fragment()) since defining one is not needed for redirects.

### CODE REVIEW

File: reflex/app.py [hunk 1]
```diff
@@ -404,10 +404,13 @@ def _check_routes_conflict(self, new_route: str):
 
     def add_custom_404_page(
         self,
-        component,
+        component=None,
```
Comment (ElijahAhianyo) [inline reflex/app.py:407]: nit: will be nice to have  annotations for all params for consistency and readability

**Review by masenf — APPROVED** [review]

File: reflex/app.py [hunk 2]
```diff
@@ -436,8 +444,17 @@ def add_custom_404_page(
...
-        froute = format.format_route
-        self.pages[froute(constants.SLUG_404)] = component
+        # format 404 route
+        route = format.format_route(constants.SLUG_404)
+
+        # add the page
+        self.pages[route] = component
+
+        # Add the load events.
+        if on_load:
```
Comment (picklelo) [inline reflex/app.py:454]: We should refactor this with the `add_page` method so we're not duplicating code. Is there any reason this method can't just call `add_page`?

**Review by picklelo — APPROVED** [review]
nice!

File: reflex/app.py [hunk 3]
```diff
@@ -404,11 +404,14 @@ def _check_routes_conflict(self, new_route: str):
 
     def add_custom_404_page(
         self,
-        component,
-        title=None,
-        image=None,
-        description=None,
-        meta=constants.DEFAULT_META_LIST,
+        component: Optional[Union[Component, ComponentCallable]] = None,
```
Comment (picklelo) [inline reflex/app.py:407]: Why is the component optional if they're adding a page?

File: reflex/app.py [hunk 4]
```diff
@@ -420,25 +423,23 @@ def add_custom_404_page(
...
-        compiler_utils.add_meta(
-            component,
-            title=title,
-            image=image,
-            description=description,
+        self.add_page(
+            component=(
+                component
+                if isinstance(component, Component)
+                else (component() if component else Fragment.create())
```
Comment (picklelo) [inline reflex/app.py:433]: This check is already done in `add_page`, so we don't need to do it here actually.

File: reflex/app.py (hunk 3 above)
Comment (Lendemor) [reply reflex/app.py:407]: If we setup a redirect, the actual page to render doesn't matter (and will be set to default as a rx.fragment()

File: reflex/app.py (hunk 4 above)
Comment (Lendemor) [reply reflex/app.py:433]: We need to for the case where component is None, maybe I can simplify it here.

**Review by picklelo — APPROVED** [review]

**Pull request merged**